### ISSUE TITLE
"Cannot send an empty message".

### ISSUE THREAD

**Issue by DaVince89** [utarwyn/discord-tictactoe#41]
Hi. I installed the discord-tictactoe game and used the costom command handler. So far it works fine, but it's only possible to play against the AI.

When I mention an user to play the game with, I get the error; "cannot send an empty message".
I can not find the problem. I did not change anything in the code, and just exactly followed the steps in "READ ME". Did I forgot something or did something wrong?

I have basic knowledge about coding bots for discord, so I am still kinda new to this.

**Comment by utarwyn** [comment]
Hello @DaVince89, it's a common error which occurs when you are using wrong version of Node.js or Discord.js. I don't know why a lot of users has this bug when using my module, if someone can explain me.. Check that you are using **Node.js 14+** with **Discord.js 12.5+** and reinstall my module if that does not work.

### More technically..

I am using an embed message to display a duel request between two players, that is why you don't have this error when playing against the AI. For an obscur reason, Discord.js is sometimes complaining where this embed is passed alone to the `send()` method. A certain amount of issues have been opened on Discord.js repo to report that problem, like so: discordjs/discord.js#1463. I will do a change by using the syntax given in the issue, I hope it will fix the problem.

https://github.com/utarwyn/discord-tictactoe/blob/301eab4e26b70b5a4802be8011c14215d3671713/src/bot/request/DuelRequestMessage.ts#L64

**Comment by DaVince89** [comment]
Thank you for your reply!

I am using node v14.17.0. So I am not sure if it is a node issue.
Is there a way I can get rid of the embed without messing up the rest of the code? 

**I also wrote a Dutch translation for this bot.**
Maybe you would like to have it.
```{
    "duel": {
      "title": ":crossed_swords: Nieuwe spel uitdaging!",
      "challenge": "{invited}, **{initier}** daagt je uit voor een spel!",
      "action": "Reageer op dit bericht om het spel te accepteren of te weigeren.",
      "expire": ":x: `{invited}` heeft te laat gereageerd!",
      "reject": ":x: `{invited}` heeft het spel geweigerd!",
      "unknown-user": "Je kan dit lid niet uitdagen!"
    },
    "game": {
      "title": ":game_die: `{player1}` **VS** `{player2}`",
      "action": "{player}, het is jouw beurt:",
      "load": "Reacties worden geladen, eventjes geduld...",
      "end": "Gelijk spel! Niemand heeft gewonnen! Nog een keer?",
      "win": ":tada: {player} geeft gewonnen!",
      "expire": ":x: Het spel is **verlopen**... Misschien is iemand niet actief?",
      "waiting-ai": ":robot: De AI is aan het spelen, eventjes geduld...",
      "ai": "AI"
    }
  }```

**Comment by utarwyn** [comment]
Thank you for the dutch translation file, I will add it!
Ugh.. so I don't understand why you get the error. Can you try to replace the quoted line in my previous comment to the following code and tell me if it works or not?
```typescript
this.message = await this.request.channel.send({ embed: this.createEmbed() }); 
```

**Comment by DaVince89** [comment]
Got this error:
```
this.message = await this.request.channel.send({ embed: this.createEmbed() });
                           ^^^^^

SyntaxError: await is only valid in async function`

ps. _(I edit a typo in the Dutch translation)_

**Comment by utarwyn** [comment]
Oh sorry! If you are editing the compiled version of the module (inside node_modules folder, js files), you should use that replacement:
```javascript
this.message = yield this.request.channel.send({ embed: this.createEmbed() });
```

**Comment by DaVince89** [comment]
Okay, I have found something that worked for me.
I got rid of the MessageEmbed and changed it to a "normal" embed. This solved the problem for me.

```
///I HAVE CHANGED THIS PART
===============================================
send() {
                return __awaiter(this, void 0, void 0, function* () {
                    this.message = yield this.request.channel.send(this.createEmbed());
                    for (const reaction of DuelRequestMessage.REACTIONS) {
                        yield this.message.react(reaction);
                    }
                    this.message
                        .awaitReactions((reaction, user) => {
                        return (DuelRequestMessage.REACTIONS.includes(reaction.emoji.name) &&
                            user.id === this.invited.id);
                    }, { max: 1, time: this.expireTime * 1000, errors: ['time'] })
                        .then(this.challengeAnswered.bind(this))
                        .catch(this.challengeExpired.bind(this));
                });
            }
===============================================
            
/// TO THIS
===============================================
send() {
    return __awaiter(this, void 0, void 0, function* () {
        var _a, _b
        this.message = yield this.request.channel.send(({embed: {
            color: ('#2980b9'),
            title: `${localize_1.default.__('duel.title')}`,
            description: `${localize_1.default.__('duel.challenge', {
                invited: this.invited.toString(),
                initier: (_b = (_a = this.request.member) === null || _a === void 0 ? void 0 : _a.displayName) !== null && _b !== void 0 ? _b : ''
            }) +
                '\n' +
                localize_1.default.__('duel.action')}`
            }}));
        for (const reaction of DuelRequestMessage.REACTIONS) {
            yield this.message.react(reaction);
        }
        this.message
            .awaitReactions((reaction, user) => {
            return (DuelRequestMessage.REACTIONS.includes(reaction.emoji.name) &&
                user.id === this.invited.id);
        }, { max: 1, time: this.expireTime * 1000, errors: ['time'] })
            .then(this.challengeAnswered.bind(this))
            .catch(this.challengeExpired.bind(this));
    });
}
===============================================
 ```

**Comment by DaVince89** [comment]
I tried your line, it works as well. Thank you!

**Comment by Toskan4134** [comment]
One question, can you make it compatible with Node v14.15.2? 
I can't update it because I use Glitch and in Glitch the allowed version is v14.15.2 and not higher.

**Comment by ToonDiscord** [comment]
> One question, can you make it compatible with Node v14.15.2?
> I can't update it because I use Glitch and in Glitch the allowed version is v14.15.2 and not higher.

Dude, don't use Glitch. It is so terrible. Use repl instead.

**Comment by Toskan4134** [comment]
> > One question, can you make it compatible with Node v14.15.2?
> > I can't update it because I use Glitch and in Glitch the allowed version is v14.15.2 and not higher.
> 
> Dude, don't use Glitch. It is so terrible. Use repl instead.

Well... the problem of Repl.it is that .json databases doesn't work at all and i have private projects not like in Repl.it so... I think is better Glitch.
All I want is the ttt compatible with Node v14.15.2

**Comment by DaVince89** [comment]
According to README, Discord-tictactoe should be compatible with node v14.X. (and higher).

I recommend you to open a new issue if you have any problems with the installation. 
I'm closing this issue since it has been solved.

**Comment by utarwyn** [comment]
I have just released version **2.1.1** with the fix! Check the full changelog here:
https://github.com/utarwyn/discord-tictactoe/releases/tag/v2.1.1

Update the package with `npm update discord-tictactoe` and you should good to go! 🥳
